stan/math/prim/mat/fun/asin.hpp

```diff
@@ -29,11 +31,23 @@ struct asin_fun {
  * @param x container
  * @return Arcsine of each variable in the container, in radians.
  */
-template <typename T>
-inline typename apply_scalar_unary<asin_fun, T>::return_t asin(const T& x) {
+template <typename T, typename = require_not_eigen_vt<std::is_arithmetic, T>>
+inline auto asin(const T& x) {
   return apply_scalar_unary<asin_fun, T>::apply(x);
 }
 
+/**
+ * Version of asin() that accepts Eigen Matrix ar matrix expressions.
```

Comment:
The typo "ar matrix expressions" is repeated in a few files.

Reply:
done